include/dos_inc.h

```diff
@@ -292,45 +289,23 @@ constexpr PhysPt assert_macro_args_ok()
 #define SGET_DWORD(s, f)                                                       \
 	mem_readd(VERIFY_SGET_ARGS(4, s, f) + pt + offsetof(s, f))
 
-/* Remains some classes used to access certain things */
-#define sOffset(s,m) ((char*)&(((s*)NULL)->m)-(char*)NULL)
-#define sGet(s,m) GetIt(sizeof(((s *)&pt)->m),(PhysPt)sOffset(s,m))
-#define sSave(s,m,val) SaveIt(sizeof(((s *)&pt)->m),(PhysPt)sOffset(s,m),val)
-
 class MemStruct {
 public:
-	Bitu GetIt(Bitu size,PhysPt addr) {
-		switch (size) {
-		case 1:return mem_readb(pt+addr);
-		case 2:return mem_readw(pt+addr);
-		case 4:return mem_readd(pt+addr);
-		}
-		return 0;
-	}
-	void SaveIt(Bitu size,PhysPt addr,Bitu val) {
-		switch (size) {
-		case 1:mem_writeb(pt+addr,(Bit8u)val);break;
-		case 2:mem_writew(pt+addr,(Bit16u)val);break;
-		case 4:mem_writed(pt+addr,(Bit32u)val);break;
-		}
-	}
-	void SetPt(Bit16u seg) { pt=PhysMake(seg,0);}
-	void SetPt(Bit16u seg,Bit16u off) { pt=PhysMake(seg,off);}
-	void SetPt(RealPt addr) { pt=Real2Phys(addr);}
+	MemStruct() = default;
+	MemStruct(uint16_t seg, uint16_t off) : pt(PhysMake(seg, off)) {}
+	MemStruct(RealPt addr) : pt(Real2Phys(addr)) {}
```

Comment:
Nice touches, as I see how these contribute to compile-time class initial values below!